Mathlib/Analysis/InnerProductSpace/PiL2.lean

```diff
@@ -1176,4 +1176,10 @@ theorem inner_matrix_col_col [Fintype m] (A B : Matrix m n 𝕜) (i j : n) :
     ⟪Aᵀ i, Bᵀ j⟫ₑ = (Aᴴ * B) i j := by
   simp [PiLp.inner_apply, dotProduct, mul_apply', mul_comm]
 
+/-- The matrix representation of `innerSL 𝕜 x` given by an orthonormal basis `b` is equal to
+the row `star (b.repr x)`). -/
+theorem toMatrix_innerSL_apply [DecidableEq ι] (b : OrthonormalBasis ι 𝕜 E) (x : E) :
+    (innerSL 𝕜 x).toMatrix b.toBasis (.singleton Unit 𝕜) = replicateRow Unit (star (b.repr x)) := by
```

Comment:
Here it would again be best to generalize to an arbitrary basis; or at the very least, generalize to an arbitrary `Subsingleton` type rather than `Unit`.